### PR TITLE
e2e: kubevirt: enforce max 2 seconds downtime during live migration

### PR DESCRIPTION
**Pull request by qinqon**
## Description

Instead of only checking the last iperf3 log line after live migration,
use iperf3 timestamps to inspect all intervals from migration start onward
and enforce a maximum of 2 seconds of zero traffic.

A new `iperf.LogDowntime(log, startTime)` helper parses lines with epoch
timestamps, counts max consecutive zero-traffic seconds after `startTime`,
and returns an error if traffic is still down. iperf3 is started with
`--forceflush --timestamps='%s '` to produce per-second lines with Unix
timestamps.

Post-migration checks poll `LogDowntime` and fail if downtime exceeds 2s.

## Checks
- [ ] My code requires changes to the documentation
- [ ] if so, I have updated the documentation as required
- [x] My code requires tests
- [x] if so, I have added and/or updated the tests as required
- [x] All the tests have passed in the CI

## How to verify

PR includes unit tests

## Summary by CodeRabbit

* **New Features**
  * Timestamp-based downtime analysis for iperf traffic monitoring during VM live migrations.
  * Propagation of migration start timestamps to traffic checks for timing-aware verification.
  * Extended timeouts and polling intervals with stricter downtime thresholds for more robust verification.

* **Tests**
  * Added comprehensive tests for downtime calculation covering multiple scenarios and edge cases.

### CODE REVIEW

File: go-controller/pkg/testing/iperf/iperf_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+				iperfLine(baseEpoch, "0.00-1.00", "200 MBytes", "1.68 Gbits/sec"),
+				iperfZeroLine(baseEpoch+1, "1.00-2.00"),
+				iperfZeroLine(baseEpoch+2, "2.00-3.00"),
+				iperfZeroLine(baseEpoch+3, "3.00-4.00"),
+				iperfLine(baseEpoch+4, "4.00-5.00", "180 MBytes", "1.51 Gbits/sec"),
+			}, "\n"),
+			startTime:        time.Unix(0, 0).UTC(),
+			expectedDowntime: 3,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (qinqon) [reply go-controller/pkg/testing/iperf/iperf_test.go:163]: Fixed

File: test/e2e/kubevirt.go [hunk 2]
```diff
@@ -329,48 +330,45 @@ var _ = Describe("Kubevirt Virtual Machines", feature.VirtualMachineSupport, fun
...
-				if len(iperfLogLines) == 0 {
-					return "", nil
-				}
-				lastIperfLogLine := iperfLogLines[len(iperfLogLines)-1]
-				return lastIperfLogLine, nil
+				g.Expect(err).NotTo(HaveOccurred(), stage+": failed reading iperf3 log")
+				g.Expect(iperfLog).NotTo(ContainSubstring("iperf3: error"), stage+": "+iperfLogFile)
+
+				maxDowntime, err := iperftest.LogDowntime(iperfLog, *startTime)
+				g.Expect(err).NotTo(HaveOccurred(), stage+": traffic should have resumed")
```
[the remark this answers is not in the document]
Comment (qinqon) [reply test/e2e/kubevirt.go:365]: Fixed

File: go-controller/pkg/testing/iperf/iperf.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+// the maximum number of consecutive seconds with zero traffic since startTime.
+// Lines before startTime are ignored. Lines without a valid epoch timestamp or
+// without " sec " are skipped.
+func LogDowntime(iperfLog string, startTime time.Time) (int, error) {
+	lines := strings.Split(strings.TrimSuffix(iperfLog, "\n"), "\n")
+
+	lastLine := lines[len(lines)-1]
+	if strings.Contains(lastLine, TrafficDownIperfLog) {
+		return 0, fmt.Errorf("traffic is still down")
+	}
```
[the remark this answers is not in the document]
Comment (qinqon) [reply go-controller/pkg/testing/iperf/iperf.go]: Fixed

**Comment by qinqon** [comment]
@coderabbitai review

**Comment by booxter** [comment]
Do we have data on actual downtime observed in CI?

Context: I remember when we added 5s downtime in OpenStack CI for the same live migration workflow, it introduced sporadic CI instability because while 99% of runs were well below the threshold, some - due to slowliness in underlying test infra - were failing. I hope this threshold enforcement won't do the same for OVN-K CI.

**Comment by qinqon** [comment]
> Do we have data on actual downtime observed in CI?
> 
> Context: I remember when we added 5s downtime in OpenStack CI for the same live migration workflow, it introduced sporadic CI instability because while 99% of runs were well below the threshold, some - due to slowliness in underlying test infra - were failing. I hope this threshold enforcement won't do the same for OVN-K CI.

This is a hard requirement for virt, if we go beyond 2 seconds the feature is broken, this PR just ensure that the 2 seconds enforcement is really checked so we don't suffer false possitives.

**Comment by qinqon** [comment]
Closing this PR.
